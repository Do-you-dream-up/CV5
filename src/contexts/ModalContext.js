import PropTypes from 'prop-types';
import React, { useState } from 'react';

const OPTIONS = { dismissable: true, variant: 'center' };

export const ModalContext = React.createContext();
export function ModalProvider({ children }) {
  const [Component, setComponent] = useState(null);
  const [onReject, setOnReject] = useState(null);
  const [onResolve, setOnResolve] = useState(null);
  const [options, setOptions] = useState(OPTIONS);
  const [thinking, setThinking] = useState(false);

  /**
   * Initialize values for a modal element. Values are further utilized by the
   * <Modal /> component.
   *
   * If an action is provided, resolve the modal only after the promised
   * returned by ACTION is fulfilled.
   *
   * @param {function} Component - The component to wrap with.
   * @param {action} [action] - Function returning a Promise object.
   * @param {Object} [options] - Extra options to pass to the modal component.
   * @returns {Promise}
   */
  const modal = (Component, action, options) =>
    new Promise((resolve, reject) => {
      setComponent(() => Component);
      setOnReject(() => onCleanup(reject));
      setOnResolve(() => (data) => {
        if (typeof action === 'function') {
          setThinking(true);
          setTimeout(
            () =>
              action(data).then(
                (response) => onCleanup(resolve)(response),
                (response) => onCleanup(reject)(response),
              ),
            1000,
          );
        } else {
          onCleanup(resolve)();
        }
      });
      setOptions((previous) => ({ ...previous, ...options }));
    });

  const onCleanup = (callback) => (data) => {
    setComponent(null);
    setOnReject(null);
    setOnResolve(null);
    setOptions(OPTIONS);
    setThinking(false);
    if (typeof callback === 'function') {
      callback(data);
    }
  };

  return (
    <ModalContext.Provider
      children={children}
      value={{
        Component,
        modal,
        onReject,
        onResolve,
        options,
        thinking,
      }}
    />
  );
}

ModalProvider.propTypes = {
  children: PropTypes.node,
};
