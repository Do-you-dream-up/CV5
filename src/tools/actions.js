const RE_ACTION =
  /javascript:\s*([\w.]+\s*\(\s*(?:'.*?'|".*?"|[^;]*)?(?:,\s*.*)*\))(?:\s*;+\s*([\w.]+\s*\(\s*(?:'.*?'|".*?"|[^;]*)?(?:,\s*.*)*\)))*/g;
const RE_ACTION_NAME = /^(\w+(?:.\w+)*)\s*\(\s*(.+)?\)/;
const RE_ACTION_PARAMETERS = /(?:^|,)\s*((('|").*?(\3))|([^,]+))/g;
const RE_ACTION_SANITIZE_STRING = /^('|")(.+)(\1)$/;

/**
 * Sanitize actions.
 */
export default (data) => {
  let actions = [];
  let response = [];
  let match = null;
  while ((match = RE_ACTION.exec(data))) {
    const [, ...matches] = match;
    actions = [...actions, ...matches.filter((it) => it)];
  }
  actions.map((it) => {
    let [, action, values] = it.match(RE_ACTION_NAME);
    let parameters = [];
    let match = null;
    while ((match = RE_ACTION_PARAMETERS.exec(values))) {
      parameters = [...parameters, match[0]];
    }
    parameters = parameters.map((it) => {
      const match = it.match(RE_ACTION_SANITIZE_STRING);
      return match ? match[2] : it;
    });
    response = [...response, { action, parameters }];
  });
  return response;
};
