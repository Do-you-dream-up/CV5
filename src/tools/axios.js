export const axiosConfigNoCache = {
  headers: {
    'Cache-Control': 'no-cache',
    Expires: '0',
  },
  params: {
    t: new Date().getTime(),
  },
};
