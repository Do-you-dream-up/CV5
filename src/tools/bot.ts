import { Local } from './storage';
import { getResourceWithoutCache } from './resources';
import { isDefined } from './helpers';

/**
 * - Wait for the bot ID and the API server then create default API based on the server ;
 * - Use BOT from local storage for the chatbox preview in Channels ;
 * - Protocol http is used when bliss is used in local with Channels.
 */

interface BotConfig {
  id: string;
  servers: string[];
  configId: string;
  getServer: (index: number) => string | null;
  getNextServer: (index: number, previousIndexUsed?: number) => string | null;
}

const isChannels = Local.isChannels.load();

export let BOT: BotConfig = {
  id: '',
  servers: [],
  configId: '',
  getServer: () => {
    return null;
  },
  getNextServer: () => {
    return null;
  },
};

export const getNextServerFromList = (servers: string[] | undefined, index: number, previousIndexUsed?: number) => {
  if (servers) {
    if (servers.length === 1) {
      return null;
    } else {
      let currentIndex: number = index;
      if (previousIndexUsed != undefined && previousIndexUsed !== index) {
        currentIndex = previousIndexUsed;
      }

      const nextIndex = currentIndex + 1;
      if (nextIndex >= servers.length) {
        return servers[0];
      }

      return servers[nextIndex];
    }
  }

  return null;
};

export const initBotInfoFromJsonOrChannels = async () => {
  const { data } = await getResourceWithoutCache('override/bot.json');

  let botData = {
    ...data,
    ...(isChannels && {
      id: Local.botId.load(),
      servers: Local.servers.load(),
    }),
  };

  const getServer = (index: number): string | null => {
    const servers: string[] = botData?.servers;
    return servers?.length > index && index >= 0 ? servers[index] : null;
  };

  const getNextServer = (index: number, previousIndexUsed?: number): string | null => {
    return getNextServerFromList(botData?.servers, index, previousIndexUsed);
  };

  botData = {
    ...botData,
    getServer: getServer,
    getNextServer: getNextServer,
  };

  BOT = Object.assign({}, botData);
  Local.botId.save(botData.id);
};
