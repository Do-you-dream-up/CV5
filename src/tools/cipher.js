import { Base64 } from 'js-base64';

/**
 * Deep-apply manipulator onto data.
 *
 * @param {*} data - Parameter description.
 * @param {function} manipulator - Function to transform with.
 * @returns {*} The transformed data.
 */
const transform = (data, manipulator) => {
  if (Array.isArray(data)) {
    for (let i = 0; i < data.length; i++) {
      data[i] = transform(data[i], manipulator);
    }
  } else if (typeof data === 'object') {
    for (let key in data) {
      data[key] = transform(data[key], manipulator);
    }
  } else if (typeof data === 'string') {
    return manipulator(data);
  }
  return data;
};

export const decode = (data) => transform(data, Base64.decode);
export const encode = (data) => transform(data, Base64.encode);
