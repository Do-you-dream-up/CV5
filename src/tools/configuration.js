import axios from 'axios';
import React from 'react';
import { ConfigurationContext } from '../contexts/ConfigurationContext';
import json from './configuration.json';
import { Local } from './storage';

/**
 * Helper class to find values in a JSON configuration file.
 */
export const configuration = new (class Configuration {
  constructor() {
    this.configuration = {};
  }

  /**
   * Fetch the configuration JSON and save its content.
   *
   * @param {string} [path] - Configuration file path.
   * @returns {Promise}
   */
  initialize = (
    path = `${process.env.PUBLIC_URL}override/configuration.json`,
  ) => {
    this.configuration = JSON.parse(JSON.stringify(json));

    return axios.get(path).then(({ data }) => {
      if (Local.get(Local.names.wizard)) {
        this.configuration = Local.get(Local.names.wizard);
        return this.sanitize(JSON.parse(JSON.stringify(this.configuration)));
      } else {
        return this.sanitize(JSON.parse(JSON.stringify(data)));
      }
    });
  };

  /**
   * Clean up and ensure sane style values. This typically will convert styles
   * into values JSS can understand.
   *
   * @param {Object} data - Configuration object to sanitize.
   * @returns {Object}
   */
  sanitize = (data) => {
    const forgeStyles = ({
      background,
      bottom,
      height,
      left,
      maxHeight,
      position,
      right,
      top,
      width,
    } = {}) => ({
      ...(background !== undefined && { background }),
      ...(bottom !== undefined && { bottom }),
      ...(height !== undefined && { height }),
      ...(left !== undefined && { left }),
      ...(maxHeight !== undefined && { maxHeight }),
      ...(position !== undefined && { position }),
      ...(right !== undefined && { right }),
      ...(top !== undefined && { top }),
      ...(width !== undefined && { width }),
    });
    Object.keys(data).forEach((key) => {
      if (data[key] instanceof Object) {
        data[key] = {
          ...data[key],
          ...(data[key].styles && { styles: forgeStyles(data[key].styles) }),
          ...(data[key].stylesMobile && {
            stylesMobile: forgeStyles(data[key].stylesMobile),
          }),
        };
      }
    });
    return data;
  };
})();

/**
 * High-order component to pass on configuration.
 */
export const withConfiguration = (Component) =>
  class InnerComponent extends React.PureComponent {
    static contextType = ConfigurationContext;
    render() {
      return (
        <Component configuration={this.context.configuration} {...this.props} />
      );
    }
  };
