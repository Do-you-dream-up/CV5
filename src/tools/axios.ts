import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
import { hasProperty, isDefined, isPositiveNumber, secondsToMs } from './helpers';
import { BOT, initBotInfoFromJsonOrChannels } from './bot';
import { getOidcEnableWithAuthStatus } from './oidc';
import Storage from '../components/auth/Storage';
import { decode } from './cipher';
import debounce from 'debounce-promise';
import dydu from './dydu';
import { Local } from './storage';

interface AxiosDyduConfig {
  server: string;
  timeout: number;
  axiosConf: any;
}

const sessionCurrentServerIndex = sessionStorage.getItem('dydu.server');
let currentServerIndex: number = sessionCurrentServerIndex ? Number.parseInt(sessionCurrentServerIndex) : 0;
let lastStatus = 'OK';
const minTimeoutForAnswer: number = secondsToMs(3);
let triesCounter = 0;
const maxTries = 3;

let configuration: any = null;
let lastResponse: any = null;

let callTokenRefresher: any = null;
let callOidcLogin: any = null;

const getAxiosInstanceWithDyduConfig = (config: AxiosDyduConfig) => {
  if (!isDefined(config?.axiosConf)) config.axiosConf = {};

  const instance = axios.create({
    baseURL: config?.server,
    timeout: isPositiveNumber(config?.timeout) ? config.timeout : axios.defaults.timeout,
    ...config.axiosConf,
  });

  // when request is sent
  instance.interceptors.request.use(
    (config: AxiosRequestConfig) => {
      if (getOidcEnableWithAuthStatus()) {
        const tokenInfo = Storage.loadToken();
        const idToken: string | null | undefined = isDefined(tokenInfo?.id_token) ? tokenInfo.id_token : undefined;
        const accessToken: string | null | undefined = isDefined(tokenInfo?.access_token)
          ? tokenInfo.access_token
          : undefined;

        if (!config.headers) {
          config.headers = {};
        }

        if (idToken) {
          config.headers['Authorization'] = `Bearer ${idToken}`;
        }

        if (accessToken) {
          config.headers['OIDCAccessToken'] = `${accessToken}`;
        }
      }
      return config;
    },
    (error) => {
      return Promise.reject(error);
    },
  );

  // when response code in range of 2xx
  const onSuccess = (response: any) => {
    return response;
  };

  instance.interceptors.response.use(onSuccess);

  return instance;
};

const handleSwitchToBackup = (currentServerPreviouslyUsed?: number) => {
  if (lastStatus === 'Error') {
    const backupServer: string | null = BOT.getNextServer(currentServerIndex, currentServerPreviouslyUsed);
    console.log('Status error, switching to server=' + backupServer);

    if (backupServer) {
      currentServerIndex++;
      if (currentServerIndex >= BOT.servers.length) {
        currentServerIndex = 0;
        triesCounter++;
        console.log('TriesCounter=' + triesCounter + ', waiting 5000ms');
        (async () => wait(5000))();
      }
      SERVLET_API.defaults.baseURL = buildServletApiUrl(backupServer);
      SERVLET.defaults.baseURL = buildServletUrl(backupServer);
    } else {
      triesCounter++;
      (async () => wait(5000))();
    }
  }
};

const wait = async (ms: number) => {
  await new Promise((resolve) => setTimeout(resolve, ms));
};

const handleSetApiTimeout = (ms: number) => {
  let timeout = minTimeoutForAnswer;
  if (ms) {
    timeout = ms;
  }

  if (SERVLET_API?.defaults) {
    SERVLET_API.defaults.timeout = timeout;
  }

  if (SERVLET?.defaults) {
    SERVLET.defaults.timeout = timeout;
  }
};

const handleAxiosResponse = (data: any) => {
  if (!data) {
    data = {};
  }

  data && configuration?.saml?.enable && samlRenewOrReject(data);

  if (hasProperty(data, 'values')) {
    data.values = decode(data.values);
    if (data.values.contextId) {
      dydu.setContextId(data.values.contextId);
    }
    return data.values;
  }

  return data;
};

const samlRenewOrReject = ({ type, values }) => {
  switch (type) {
    case 'SAML_redirection':
      redirectAndRenewAuth(values);
      break;
    default:
      return renewAuth(values?.auth);
  }
};

const redirectAndRenewAuth = (values: any) => {
  const relayState = encodeURI(window.location.href);
  // const relayState = JSON.stringify({ redirection: encodeURI(window.location.href), bot: BOT.id });
  try {
    renewAuth(atob(values?.auth));
    window.location.href = `${atob(values?.redirection_url)}&RelayState=${relayState}`;
  } catch {
    renewAuth(values?.auth);
    window.location.href = `${values?.redirection_url}&RelayState=${relayState}`;
  }
};

const renewAuth = (auth: string) => {
  if (auth) {
    try {
      Local.saml.save(atob(auth));
    } catch {
      Local.saml.save(auth);
    }
  }
};

const handleAxiosError = (
  error: any,
  verb: any,
  path: string,
  data: any,
  timeout: number,
  currentServerIndexForThisRequest: number,
) => {
  if (triesCounter >= maxTries) {
    throw 'API Unreachable';
  }

  /**
   * NO 401 ERROR
   */
  if (error?.response?.status !== 401) {
    lastStatus = 'Error';
  }

  /**
   * IF 401
   */
  if (getOidcEnableWithAuthStatus()) {
    return new Promise((resolve) =>
      handleTokenRefresh(() => {
        setTimeout(() => {
          return resolve(emit(verb, path, data, timeout, currentServerIndexForThisRequest));
        }, minTimeoutForAnswer);
      }),
    );
  }

  // Retry API Call
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(emit(verb, path, data, timeout, currentServerIndexForThisRequest));
    }, minTimeoutForAnswer);
  });
};

const handleTokenRefresh = (retry: any) => {
  if (configuration?.oidc?.enable) {
    const tokenInfo = Storage.loadToken();
    const refreshToken = isDefined(tokenInfo?.refresh_token) ? tokenInfo.refresh_token : undefined;

    if (refreshToken && refreshToken?.length > 0 && triesCounter < 2 && Storage.loadPkceCodeVerifier()) {
      console.log('Refreshing token...');
      callTokenRefresher().then(() => {
        return retry();
      });
    } else {
      console.log('No refresh token found, redirecting to login page...');
      Storage.clearToken();
      callOidcLogin();
      return Promise.resolve();
    }
  }
};

const formatConsoleError = (error: any) => {
  console.group('An error occurred while sending the request to Dydu. Please send us this group of logs :');
  console.log('Date : ', new Date());
  console.log('Error : ', error.message);
  console.log('Status : ', error.status || error?.response?.status);
  console.log('Request : [%s] %s%s', error.config.method.toUpperCase(), error.config.baseURL, error.config.url);
  console.log('Headers :', JSON.stringify(error.config.headers));
  console.log(
    'Data :\n%s',
    error.config.data
      ? error.config.data
          .trim()
          .split('&')
          .map((s: string) => decodeURIComponent(s))
          .join('\n')
      : 'no data',
  );
  console.groupEnd();
};

/**
 * Request against the provided path with the specified data. When
 * the response contains values, decode it and refresh the context ID.
 * if the request fail several times the request will be failover to back-up server.
 *
 * @param {function} verb - A verb method to request with.
 * @param {string} path - Path to send the request to.
 * @param {Object} data - Data to send.
 * @param {number} timeout - timeout value to use.
 * @param {boolean} ignoreSwitch - set tu true, the server switch will not occur.
 * @param {number} currentServerPreviouslyUsed - server index used with the last request. Used to switch server
 * @returns {Promise}
 */
const emit = debounce(
  (verb: any, path: string, data: any, timeout: number, currentServerPreviouslyUsed?: number) => {
    handleSwitchToBackup(currentServerPreviouslyUsed);
    handleSetApiTimeout(timeout);

    const pathWithoutTrailingSlash = path.replace(/\/$/, '');

    const currentServerIndexForThisRequest = currentServerIndex;

    try {
      return verb(pathWithoutTrailingSlash, data)
        .then((response: any) => {
          lastResponse = response;
          lastStatus = 'OK';
          triesCounter = 0;
          sessionStorage.setItem('dydu.server', currentServerIndex.toString());
          return response;
        })
        .then(({ data = {} }) => {
          return handleAxiosResponse(data);
        })
        .catch((error: any) => {
          formatConsoleError(error);
          return handleAxiosError(
            error,
            verb,
            pathWithoutTrailingSlash,
            data,
            timeout,
            currentServerIndexForThisRequest,
          );
        });
    } catch (e) {
      console.error('while executing |emit()|', e);
    }
  },
  100,
  { leading: true },
);

const getServerFromIndex = (index: number): string | null => {
  return BOT !== null ? BOT.getServer(index) : null;
};

const isLocalServer = (server: string | null): boolean => {
  return (
    server !== null &&
    (server.includes('dev.dydu.local') || server.includes('localhost') || server.endsWith('.loca.lt'))
  );
};

const buildServletUrl = (server?: string | null): string => {
  if (!server) {
    server = getServerFromIndex(currentServerIndex);
  }

  const servletPath = isLocalServer(server) ? '' : 'servlet/';
  return `${server}/${servletPath}`;
};

const buildServletApiUrl = (server?: string | null): string => {
  return buildServletUrl(server) + 'api/';
};

let SERVLET_API: AxiosInstance, SERVLET: AxiosInstance;

initBotInfoFromJsonOrChannels().then(() => {
  SERVLET_API = getAxiosInstanceWithDyduConfig({
    server: buildServletApiUrl(),
    timeout: minTimeoutForAnswer,
    axiosConf: {
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
      },
    },
  });

  /**
   * Used for old Servlet API
   */
  SERVLET = getAxiosInstanceWithDyduConfig({
    server: buildServletUrl(),
    timeout: minTimeoutForAnswer,
    axiosConf: {
      headers: {
        Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
        'Content-Type': 'multipart/form-data',
      },
    },
  });
});

const getLastResponse = (): any => lastResponse;
const setLastResponse = (newLastResponse: any) => (lastResponse = newLastResponse);

const setConfiguration = (newConfiguration: any) => (configuration = newConfiguration);

const setCallTokenRefresher = (newCallTokenRefresher: any) => {
  callTokenRefresher = newCallTokenRefresher;
};

const setCallOidcLogin = (newCallOidcLogin: any) => {
  callOidcLogin = newCallOidcLogin;
};

const emitLivechatError = () => {
  currentServerIndex++;
};

export {
  getLastResponse,
  setLastResponse,
  setConfiguration,
  setCallTokenRefresher,
  setCallOidcLogin,
  getAxiosInstanceWithDyduConfig,
  emit,
  emitLivechatError,
  buildServletUrl,
  SERVLET_API,
  SERVLET,
  currentServerIndex,
};
