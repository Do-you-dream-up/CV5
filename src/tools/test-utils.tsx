import { JssProvider, ThemeProvider } from 'react-jss';
import React, { ReactElement } from 'react';
import { RenderOptions, render } from '@testing-library/react';

import { ConfigurationProvider } from '../contexts/ConfigurationContext';
import { EventsProvider } from '../contexts/EventsContext';
import ViewModeProvider from '../contexts/ViewModeProvider';
import configuration from '../../public/override/configuration.json';
import theme from '../../public/override/theme.json';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <JssProvider>
      <ThemeProvider theme={theme}>
        <ConfigurationProvider configuration={configuration}>
          <ViewModeProvider>
            <EventsProvider>{children}</EventsProvider>
          </ViewModeProvider>
        </ConfigurationProvider>
      </ThemeProvider>
    </JssProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
