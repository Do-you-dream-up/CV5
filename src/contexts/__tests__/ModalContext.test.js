import '@testing-library/jest-dom';

import { ModalContext, ModalProvider } from '../ModalContext';

import PropTypes from 'prop-types';
import React from 'react';
import { render } from '../../tools/test-utils';

describe('ModalProvider', () => {
  it('renders children without crashing', () => {
    const { container } = render(
      <ModalProvider>
        <div>Child Component</div>
      </ModalProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('provides modal context values to children', () => {
    const ChildComponent = () => {
      const { Component, modal, onReject, onResolve, options, thinking } = React.useContext(ModalContext);
      return (
        <>
          <p>Component: {Component ? 'open' : 'closed'}</p>
          <p>Thinking: {thinking ? 'true' : 'false'}</p>
          <button data-testid="modal-button-open" onClick={() => modal(<div>Test Modal</div>)}>
            Open Modal
          </button>
          <button data-testid="modal-button-resolve" onClick={onResolve}>
            Resolve Modal
          </button>
          <button data-testid="modal-button-reject" onClick={onReject}>
            Reject Modal
          </button>
          <p>Options: {JSON.stringify(options)}</p>
        </>
      );
    };

    ChildComponent.propTypes = {
      onButtonClick: PropTypes.func,
    };

    const { getByText, getByTestId } = render(
      <ModalProvider>
        <ChildComponent />
      </ModalProvider>,
    );

    expect(getByText('Component: closed')).toBeInTheDocument();
    expect(getByText('Thinking: false')).toBeInTheDocument();
    expect(getByTestId('modal-button-open')).toBeInTheDocument();
    expect(getByTestId('modal-button-resolve')).toBeInTheDocument();
    expect(getByTestId('modal-button-reject')).toBeInTheDocument();
    expect(getByText('Options: {"dismissable":true,"variant":"center"}')).toBeInTheDocument();

    getByTestId('modal-button-open').click();
    expect(getByText('Component: open')).toBeInTheDocument();

    getByTestId('modal-button-resolve').click();
    // expect(getByText('Component: closed')).toBeInTheDocument();
    // expect(getByText('Thinking: true')).toBeInTheDocument();

    // // Wait for the promise to resolve
    // setTimeout(() => {
    //   expect(getByText('Component: closed')).toBeInTheDocument();
    //   expect(getByText('Thinking: false')).toBeInTheDocument();
    // }, 2000);

    // getByTestId('modal-button-open').click();
    // expect(getByText('Component: open')).toBeInTheDocument();

    // getByTestId('modal-button-reject').click();
    // expect(getByText('Component: closed')).toBeInTheDocument();
    // expect(getByText('Thinking: false')).toBeInTheDocument();
  });
});
