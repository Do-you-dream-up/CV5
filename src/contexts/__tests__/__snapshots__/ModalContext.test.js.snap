// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalProvider renders children without crashing 1`] = `
<div>
  Child Component
</div>
`;
